### PR TITLE
makes mars64 seeds to be checked by a centeral code block

### PR DESCRIPTION
**Pull request by alan-stokes**
title. 

tested by https://github.com/SpiNNakerManchester/sPyNNaker8/pull/452

### CODE REVIEW

File: spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural_common.py [hunk 1]
```diff
@@ -285,17 +286,17 @@ def __write_common_rewiring_data(
...
         # write the random seed (4 words), generated randomly,
         # but the same for all postsynaptic vertices!
         for seed in self.__seeds[app_vertex]:
             spec.write_value(data=seed)
 
         # write local seed (4 words), generated randomly!
-        for _ in range(4):
-            spec.write_value(data=numpy.random.randint(0x7FFFFFFF))
+        for seed in create_mars_kiss_seeds(self.__rng, self.__seed):
+            spec.write_value(data=seed)
```
Comment (dkfellows) [inline spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural_common.py:299]: It should be possible to `spec.write_array(create_mars_kiss_seeds(self.__rng, self.__seed))`

File: spynnaker/pyNN/utilities/utility_calls.py [hunk 2]
```diff
@@ -275,18 +282,27 @@ def low(dist):
...
-        seed[1] = 13031301
+        seed[1] = ARBITRARY_Y
 
     # avoid z=c=0 and make < 698769069
-    seed[3] = seed[3] % 698769068 + 1
+    seed[3] = seed[3] % MARS_C_MAX + 1
     return seed
 
 
+def create_mars_kiss_seeds(rng, seed):
```
Comment (dkfellows) [inline spynnaker/pyNN/utilities/utility_calls.py:297]: Needs a little bit of a doc comment here.

**Review by dkfellows — CHANGES_REQUESTED** [review]

File: spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural_common.py (hunk 1 above)
Comment (alan-stokes) [reply spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural_common.py:299]: this is true. will fix

Comment (alan-stokes) [reply spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural_common.py]: fixed

File: spynnaker/pyNN/utilities/utility_calls.py (hunk 2 above)
Comment (alan-stokes) [reply spynnaker/pyNN/utilities/utility_calls.py:297]: oops. fixing

Comment (alan-stokes) [reply spynnaker/pyNN/utilities/utility_calls.py:297]: doced

**Review by rowleya — APPROVED** [review]

**Pull request merged**